samples/azure-quantum/CONTRIBUTING.md

```diff
@@ -0,0 +1,40 @@
+# Process for creating notebook samples for the Azure Quantum portal experience
+This document focuses on the process for creating samples that will eventually appear in the hosted notebooks sample gallery in the portal. If you are creating a sample for other purposes or means of consumption, then this document might not apply to your scenario. (If you are interested in contributing an optimization notebook sample, please see [this repository](https://github.com/microsoft/qio-samples)).
+
+There are three steps involved in creating a new sample.
+
+1. Write the notebook sample
+1. Test it in the portal
+1. Merge it into `main` AND into `feature/samples-gallery`
```

Comment:
Another important point: making sure all appropriate metadata is there for inclusion in aka.ms/try-qsharp and docs.microsoft.com/samples. For the former, that's mostly binder-index.md and is largely enforced via CI. For the latter, that consists of making sure there's an appropriate YAML header on each README.md file.

It may help to list a prototypical structure for a new notebook sample. E.g.: something roughly like the following.

```
binder-index.md # Should link to subject area and each way of using each sample
samples/
  subject-area/
     README.md # should describe subject area and link to constituent samples
     sample-name/
         README.md # should have YAML header for docs.ms/samples and manifest section
         sample-name.ipynb
         sample-name.csproj # Not yet supported in portal context.
```